internal/consumer/partitionConsumer.go

```diff
@@ -385,6 +391,9 @@ func (p *rangePartitionConsumer) offsetLoop() {
 				p.Drain(p.options.MaxProcessingTime)
 				return
 			}
+			p.offsetRangeLock.Lock()
+			p.offsetRange = &ntf
+			p.offsetRangeLock.Unlock()
```

Comment:
maybe enclose the thread safe write and read on p.offsetRange in separate method call.